src/main/java/org/matsim/episim/InfectionEventHandler.java

```diff
@@ -630,15 +634,19 @@ public void reset(int iteration) {
 		for (EpisimPerson person : personMap.values()) {
 			progressionModel.updateState(person, iteration);
 		}
-
+		reporting.reportCpuTime(iteration, "ProgressionModelParallel", "start", -2);
+		progressionModel.afterStateUpdates(personMap, iteration);
+		reporting.reportCpuTime(iteration, "ProgressionModelParallel", "finished", -2);
 		reporting.reportCpuTime(iteration, "ProgressionModel", "finished", -1);
 
 		reporting.reportCpuTime(iteration, "VaccinationModel", "start", -1);
 		int available = EpisimUtils.findValidEntry(vaccinationConfig.getVaccinationCapacity(), 0, date);
-		vaccinationModel.handleVaccination(personMap, false, (int) (available * episimConfig.getSampleSize()), iteration, now);
+		if (available > 0)
```

Comment:
This check is now already done inside the vaccination models. I prefer it at this location, because one model might want to calculate something at the start of the day, even though there is no capacity.